File: medium/twoSum/twoSum.java
import java.util.Scanner;
import java.util.HashMap;
import java.util.Arrays;

public class twoSum {
	public static void main(String[]args) {
		Scanner scan = new Scanner(System.in);
		System.out.print("Enter a series of integers separated by spaces: ");
		String[] input = scan.nextLine().split(" ");
		System.out.print("Enter the integer sum to search for: ");
		int target = scan.nextInt();

		int[] nums = new int[input.length];
		for (int i = 0; i < input.length; i++) {
			nums[i] = Integer.parseInt(input[i]);
		}

		System.out.println(Arrays.toString(twoSum(nums, target)));
	}

	public static int[] twoSum(int[] nums, int target) {
		HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
		for (int i = 0; i < nums.length; i++) {
			if (hash.containsKey(target-nums[i])) {
				return new int[]{hash.get(target-nums[i]),i};
			} else {
				hash.put(nums[i], i);
			}
		}
		return new int[]{0,0};
	}
}

File: medium/twoSum/twoSum.py
# Makes Python 2.x and 3.x play nice.
try: input = raw_input
except NameError: pass

def twoSum(nums, target):
    dict = {}
    for i in range(len(nums)):
        if target-nums[i] in dict:
            return [dict[target-nums[i]],i]
        else:
            dict[nums[i]] = i
    return [0,0]

print(twoSum(eval(input("Enter an array of integers: ")),int(input("Enter a target integer: "))))